### ISSUE TITLE
Drop of ControlFlow into an Activity diagram does not automatically connect the arrow to the source and target objects in the diagram

### ISSUE THREAD

**Issue by pbrown12303** [gaphor/gaphor#3477]
### Describe the bug
When programmatically creating an activity diagram, if you drop a ControlFlow that has its source and target set into the diagram and the diagram already contains images of the source and target elements, the control flow image is not automatically connected to  the source and target images.

### To Reproduce
1. Create an Activity and an Activity Diagram programmatically
2. Create two actions, A and B.
3. Drop actions A and B onto the diagram
4. Create a ControlFlow. Set its source and target to actions A and B.
5. Drop the ControlFlow onto the diagram.

### Expected behavior
The ControlFlow should connect automatically to the end-point images.

### OS
- [ ] Linux (Please put in notes the specific distro)
- [ ] macOS
- [x] Windows

### Version
Version of Gaphor: 2.26.0

**Comment by pbrown12303** [comment]
I think I found the problem. The code that handles connecting relationships is in drop.py. The specific function is:
```
@drop.register(Relationship, Diagram)
def drop_relationship_on_diagram(element: Relationship, diagram: Diagram, x, y):
    item_class = get_diagram_item(type(element))
    if not item_class:
        return None

    metadata = get_diagram_item_metadata(item_class)
    return (
        drop_relationship(
            element,
            metadata["head"].get(element),
            metadata["tail"].get(element),
            diagram,
            x,
            y,
        )
        if metadata
        else None
    )

```
The problem is that ControlFlow and ObjectFlow are not sub-classes of Relationship - they are sub-classes of ActivityEdge. I think the fix is to make a copy of this function and change the Relationship type to ActivityEdge in the copy.

**Comment by danyeaw** [comment]
Closed by #3479.
